### PR TITLE
Add S3 connector setup instructions to README.md

### PR DESCRIPTION
**Pull request by busma13**
Updated README.md with the following additions:
- Getting Started: commands to install teraslice-cli and deploy assets
- Connectors: config options table and example
- Development: test and build instructions

### CODE REVIEW

**Comment by busma13** [comment]
I don't have a good description for the `bucketEndpoint` property. Any idea what this does?

**Comment by godber** [comment]
> I don't have a good description for the `bucketEndpoint` property. Any idea what this does?

It is passed through to the AWS S3 client library I think:

https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-middleware-bucket-endpoint/Interface/BucketEndpointInputConfig/

Though now I'm a little confused how we support two mutually exclusive booleans:

* `bucketEndpoint` - "Whether to use the bucket name as the endpoint for this request. The bucket name must be a domain name with a CNAME record alias to an appropriate virtual hosted-style S3 hostname, e.g. a bucket of images.johnsmith.net and a DNS record of:"
* `forcePathStyle` - "Whether to force path style URLs for S3 objects (e.g., https://s3.amazonaws.com/<bucketName>/<key> instead of https://<bucketName>.s3.amazonaws.com/<key>"

Do we enforce the fact that they are mutually exclusive?  I don't even seen `bucketEndpoint` in our code.

**Comment by godber** [comment]
Here is `bucketEndpoint` in terafoundation:

https://github.com/search?q=repo%3Aterascope%2Fteraslice%20bucketEndpoint&type=code

File: README.md [hunk 1]
```diff
@@ -13,6 +13,85 @@ This bundle includes the following processors:
...
+
+```bash
+# Step 1: make sure you have teraslice-cli installed
+yarn global add teraslice-cli
+
+# Step 2:
+teraslice-cli assets deploy clusterAlias terascope/file-assets
+```
+
+**IMPORTANT:** Additionally make sure have installed the required [connectors](#connectors).
```
Comment (godber) [inline README.md:32]: You should remove this comment, there's no additional connector in this case.

File: README.md [hunk 2]
```diff
@@ -13,6 +13,85 @@ This bundle includes the following processors:
...
+
+## Connectors
+### S3 Connector
+
+**Configuration:**
+
+The terafoundation level S3 configuration is as follows:
```
Comment (godber) [inline README.md:39]:
```suggestion
The S3 connector configuration, in your Teraslice configuration file, includes the following parameters:
```

File: README.md [hunk 3]
```diff
@@ -13,6 +13,85 @@ This bundle includes the following processors:
...
+
+| Configuration | Description | Type |  Notes |
+| --------- | -------- | ------ | ------ |
+| endpoint | Target S3 endpoint | String | defaults to `127.0.0.1:80`
```
Comment (godber) [inline README.md:43]:
```suggestion
| endpoint | Target S3 HTTP endpoint, must be URL | String | defaults to `http://127.0.0.1:80` |
```

Add final `|` and make a proper URL

**Review by godber — CHANGES_REQUESTED** [review]

**Review by godber — APPROVED** [review]

**Pull request merged**